git-submodules-status

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+################################################################################
+#
+#   Description:
+#   ------------
+#       Fetches all submodules from their respective remote repositories, and
+#       shows a table that lists for each submodule : the latest available
+#       version, the current version and a small visual cue to easily identify
+#       mismatches between the two.
+#
+#   Usage:
+#   ------
+#   (first place this script in one of the directories in $PATH)
+#       $> cd path/to/repo
+#       $> git submodules-status
+#
+################################################################################
+
+appendSpaces()
+{
+    INITIAL_STR=${1}
+    TARGET_LENGTH=${2}
+
+    SPACES_TO_APPEND=$((TARGET_LENGTH - ${#INITIAL_STR}))
+
+    echo -n ${INITIAL_STR}
+    for ((i = 0; i < ${SPACES_TO_APPEND}; i++))
+    do
+        echo -n " "
+    done
+    echo ""
+}
+
+# Confirm that we are in a git repository
+GIT_TOP=$(git rev-parse --show-toplevel) || exit $?
+
+# Check if the project has any submodules in the first place
+if ! [ -a "${GIT_TOP}/.gitmodules" ]; then
+    PROJECT_NAME=$(basename ${GIT_TOP})
+    echo "Project '${PROJECT_NAME}' has no submodules. Nothing to do here!"
+    exit 0
+fi
+
+# Save the present working directory as we'll be changing directories
+# (potentially several times)
+ORIG_DIR=${PWD}
+
+# cd to the top-level git directory so that submodule paths are relative to it
+cd ${GIT_TOP}
+
+# Confirm that all the submodules have been initialised
+UNINITIALISED_SUBMODS=(`git submodule status | grep "^-" | awk -F"/" '{print $NF}'`)
+if [ ${#UNINITIALISED_SUBMODS[@]} -gt 0 ]; then
+    echo "The following submodules haven't been initialized : ${UNINITIALISED_SUBMODS[@]}"
+    echo "Try running this first:"
+    echo "    git submodule update --init"
+    exit 2
+fi
+
+# Read the existing submodules information into an array. After this, the array
+# should contain an even number of entries, with each pair of entries containing
+# the path of the submodule (relative to the top-level git directory) and the
+# current submodule version (within parentheses) respectively.
+SUBMODS_INFO_ARR=(`git submodule status | sed 's/^\s\+//g' | cut -d" " -f2-`)
+
+# Fetch each submodule from the remote repo so that we can get its latest tag
+git submodule foreach git remote update -p
```

Comment:
No reason, apart from it being my personal preference when I update my repos. But you're right; there's no reason for that to be in this script. I'll remove it.